### PR TITLE
Update comparisons to resolve the strict comparison warnings

### PR DESCRIPTION
**Pull request by grantpalin**
See discussion at https://github.com/Automattic/_s/issues/784

### CODE REVIEW

File: comments.php [hunk 1]
```diff
@@ -72,7 +72,7 @@
 
 	<?php
 		// If comments are closed and there are comments, let's leave a little note, shall we?
-		if ( ! comments_open() && '0' != get_comments_number() && post_type_supports( get_post_type(), 'comments' ) ) :
+		if ( ! comments_open() && get_comments_number() && post_type_supports( get_post_type(), 'comments' ) ) :
 	?>
```
Comment (philiparthurmoore) [inline comments.php]: Why'd you take out `'0' !=` instead of adding in `'0' !==`?

Comment (obenland) [inline comments.php]: I think I started doing that in default themes at some point. We don't really need a strict comparison here, we just need to know if it's truthy.

Comment (grantpalin) [inline comments.php]: I did that based on the conclusion from https://github.com/Automattic/_s/issues/632

**Comment by davidakennedy** [comment]
Tested these out, and everything looks good.

**Comment by mtomas7** [comment]
Twenty Fifteen uses:

```if ( ! display_header_text() ) :```

instead of:

```if ( 'blank' === $header_text_color ) :```

Should we go with the 2015 method?

**Comment by davidakennedy** [comment]
@mtomas7 Probably a good thing to consider. Feel free to open an issue or pull request to get more eyes on it. :smile:

**Pull request merged**